src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -253,7 +253,9 @@ protected static void OnSysCall(ExecutionEngine engine, Instruction instruction)
         /// <param name="datoshi">The amount of GAS, in the unit of datoshi, 1 datoshi = 1e-8 GAS, to be added.</param>
         protected internal void AddFee(long datoshi)
         {
+#pragma warning disable CS0618 // Type or member is obsolete
             FeeConsumed = GasConsumed = checked(FeeConsumed + datoshi);
```

Comment:
Frankly, I don't understand why we have `FeeConsumed` here. It's perfectly fine to count `GasConsumed` in ~~s~~datoshis, that's what you use in VM/contract anyway.

Reply:
I don't know. I'm just fixing the warning. @Jim8y any comment?

Reply:
> Frankly, I don't understand why we have `FeeConsumed` here. It's perfectly fine to count `GasConsumed` in ~s~datoshis, that's what you use in VM/contract anyway.

@roman-khimov No, its not `necessarily` to use FeeConsumed here, problem is the same as anywhere else, gas is gas, datoshi is datoshi, miss using it cause ambiguous. So i tried to correct gas as much as possible. But if the team thinks we should keep using GasConsumed, it can be corrected.